keyboards/bastardkb/charybdis/charybdis.c

```diff
@@ -0,0 +1,245 @@
+/* Copyright 2020 Christopher Courtney <drashna@live.com> (@drashna)
+ * Copyright 2021 Quentin LEBASTARD <qlebastard@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Publicw License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "charybdis.h"
+#include "transactions.h"
+#include <string.h>
+#include "drivers/sensors/pmw3360.h"
+#include "print.h"
+
+#ifndef CHARYBDIS_DPI_OPTIONS
+#    define CHARYBDIS_DPI_OPTIONS \
+        { 600, 800, 1000, 1200, 1400, 1600, 1800, 2000, 2200, 2400, 2600, 2800  }
```

Comment:
I disabled cycling, you can only go up or down. This is why there is so many of them, so it can feel more like a scale rather than very different ones

Reply:
That makes sense, I should have checked the implementation of `constrain()` instead of assuming a modulo-like operation.  The large scale approach makes perfect sense now, thanks.